File: app/src/main/java/ch/threema/app/activities/VerificationLevelActivity.java
/*  _____ _
 * |_   _| |_  _ _ ___ ___ _ __  __ _
 *   | | | ' \| '_/ -_) -_) '  \/ _` |_
 *   |_| |_||_|_| \___\___|_|_|_\__,_(_)
 *
 * Threema for Android
 * Copyright (c) 2013-2025 Threema GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License, version 3,
 * as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 */

package ch.threema.app.activities;

import androidx.appcompat.app.ActionBar;
import ch.threema.app.R;
import ch.threema.base.utils.LoggingUtil;

import android.os.Bundle;
import android.view.MenuItem;

import org.slf4j.Logger;

import static ch.threema.app.utils.ActiveScreenLoggerKt.logScreenVisibility;

public class VerificationLevelActivity extends ThreemaToolbarActivity {
    private static final Logger logger = LoggingUtil.getThreemaLogger("VerificationLevelActivity");

    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        logScreenVisibility(this, logger);

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setTitle(R.string.verification_levels_title);
        }

    }

    @Override
    public int getLayoutResource() {
        return R.layout.activity_verification_level;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                break;
        }
        return false;
    }
}

File: app/src/main/java/ch/threema/app/activities/VerificationLevelActivity.kt
package ch.threema.app.activities

import android.content.Context
import android.os.Bundle
import android.view.MenuItem
import android.view.View
import ch.threema.android.buildActivityIntent
import ch.threema.app.R
import ch.threema.app.ui.InsetSides
import ch.threema.app.ui.applyDeviceInsetsAsPadding
import ch.threema.app.utils.logScreenVisibility
import ch.threema.base.utils.getThreemaLogger

private val logger = getThreemaLogger("VerificationLevelActivity")

class VerificationLevelActivity : ThreemaToolbarActivity() {

    init {
        logScreenVisibility(logger)
    }

    override fun getLayoutResource(): Int = R.layout.activity_verification_level

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val actionBar = supportActionBar
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true)
            actionBar.setTitle(R.string.verification_levels_title)
        }
    }

    override fun handleDeviceInsets() {
        super.handleDeviceInsets()
        findViewById<View>(R.id.scroll_container)?.applyDeviceInsetsAsPadding(
            insetSides = InsetSides.lbr(),
        )
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
        }
        return false
    }

    companion object {
        @JvmStatic
        fun createIntent(context: Context) = buildActivityIntent<VerificationLevelActivity>(context)
    }
}
